web/components/roadmap/MiniRoadmap.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { ReactElement, useCallback, useEffect, useState } from "react";
+import { useRoadmap } from "@/lib/data-hooks/useRoadmap";
+import { VerticalStepIndicator } from "@/components/njwds-extended/VerticalStepIndicator";
+import { MiniRoadmapTask } from "@/components/roadmap/MiniRoadmapTask";
+import { Icon } from "@/components/njwds/Icon";
+import analytics from "@/lib/utils/analytics";
+import { useUserData } from "@/lib/data-hooks/useUserData";
+import { isStepCompleted } from "@/lib/utils/helpers";
+import { SectionType } from "@/lib/types/types";
+import { SectionDefaults } from "@/display-content/roadmap/RoadmapDefaults";
+import { Accordion, AccordionSummary, AccordionDetails } from "@mui/material";
+import { getSectionNames } from "@/lib/utils/helpers";
+
+interface Props {
+  activeTaskId: string;
+  onTaskClick?: () => void;
+}
+
+export const MiniRoadmap = (props: Props): ReactElement => {
+  const { roadmap } = useRoadmap();
+  const getActiveStepNumber = useCallback(
+    () =>
+      roadmap?.steps.find((step) => step.tasks.map((it) => it.id).includes(props.activeTaskId))?.step_number,
+    [props.activeTaskId, roadmap?.steps]
+  );
+  const [activeStepNumber, setActiveStepNumber] = useState<number | undefined>(getActiveStepNumber());
+  const [openSteps, setOpenSteps] = useState<number[]>([]);
+  const { userData, update } = useUserData();
+  const roadmapExists = !!roadmap;
+  const roadmapSections = getSectionNames(roadmap);
+
+  useEffect(() => {
+    const activeStep = getActiveStepNumber();
+    setActiveStepNumber(activeStep);
+    setOpenSteps(activeStep ? [activeStep] : []);
+  }, [props.activeTaskId, roadmapExists, getActiveStepNumber]);
+
+  const toggleStep = (stepNumber: number): void => {
+    analytics.event.task_mini_roadmap_step.click.expand_contract();
+    if (openSteps.includes(stepNumber)) {
+      setOpenSteps(openSteps.filter((number) => number !== stepNumber));
+    } else {
+      setOpenSteps([...openSteps, stepNumber]);
+    }
+  };
+
+  const handleAccordionStateChange = async (
+    sectionType: SectionType,
+    expanded: boolean | undefined
+  ): Promise<void> => {
+    const openRoadMapSections = userData?.preferences.roadmapOpenSections;
+
+    if (expanded && openRoadMapSections) {
```

Comment:
stylistic preference is to have a guard clause. instead of `&& openRoadMapSections`, in both the if AND the else-if, you could instead have a line above this if that's just:
```
if (!openRoadMapSections) return;
``` 

that way, you get the following benefits:
- you can use an if/else instead of an if/else-if
- the if statements are easier to read. I had a hard time puzzle through "what does it mean, if it's expanded and the openroadmapsections exists... when would it not exist though? what happens then?" until I realized the second half was just a guard against undefined. a guard clause lets us move past that and then the if clause is descriptive "if it's expanded, do this, otherwise, do that". very straightforward and easier to read

Reply:
Addressed.